Sources/NIO/Bootstrap.swift

```diff
@@ -705,3 +705,116 @@ public final class DatagramBootstrap {
         }
     }
 }
+
+/// A `NIOPipeBootstrap` is an easy way to bootstrap a `PipeChannel` which uses two (uni-directional) UNIX pipes
+/// and makes a `Channel` out of them.
+///
+/// Example bootstrapping a `Channel` using `stdin` and `stdout`:
+///
+///     let channel = try NIOPipeBootstrap(group: group)
+///                       .channelInitializer { channel in
+///                           channel.pipeline.addHandler(MyChannelHandler())
+///                       }
+///                       .withPipes(inputDescriptor: STDIN_FILENO, outputDescriptor: STDOUT_FILENO)
+///
+public final class NIOPipeBootstrap {
+    private let group: EventLoopGroup
+    private var channelInitializer: ((Channel) -> EventLoopFuture<Void>)?
+    @usableFromInline
+    internal var _channelOptions = ChannelOptions.Storage()
+
+    /// Create a `PipeBootstrap` on the `EventLoopGroup` `group`.
+    ///
+    /// - parameters:
+    ///     - group: The `EventLoopGroup` to use.
+    public init(group: EventLoopGroup) {
+        self.group = group
+    }
+
+    /// Initialize the connected `PipeChannel` with `initializer`. The most common task in initializer is to add
+    /// `ChannelHandler`s to the `ChannelPipeline`.
+    ///
+    /// The connected `Channel` will operate on `ByteBuffer` as inbound and outbound messages. Please note that
+    /// `IOData.fileRegion` is _not_ supported for `PipeChannel`s because `sendfile` only works on sockets.
+    ///
+    /// - parameters:
+    ///     - handler: A closure that initializes the provided `Channel`.
+    public func channelInitializer(_ handler: @escaping (Channel) -> EventLoopFuture<Void>) -> Self {
```

Comment:
Isn't `Self` Swift 5.1?

Reply:
for return types, it was there for a long time